src/main/java/com/javadeobfuscator/deobfuscator/transformers/allatori/StringEncryptionTransformer.java

```diff
@@ -85,114 +110,174 @@ public boolean canCheckEquality(JavaValue first, JavaValue second, Context conte
+        isAllatori = isAllatori && TransformerHelper.containsInvokeSpecial(decryptorNode, "java/lang/String", "<init>", null);
+        return isAllatori;
+    }
+
+    private void patchMethod(MethodNode method) {
+        boolean getStackTrace = false;
+        boolean getClassName = false;
+        for (AbstractInsnNode i = method.instructions.getFirst(); i != null; i = i.getNext()) {
+            if (!(i instanceof MethodInsnNode)) {
+                continue;
+            }
+            String name = ((MethodInsnNode) i).name;
+            if (!getStackTrace && name.equals("getStackTrace")) {
+                getStackTrace = true;
+                if (getClassName) {
+                    break;
+                }
+            } else if (!getClassName && name.equals("getClassName")) {
+                getClassName = true;
+                if (getStackTrace) {
+                    break;
+                }
+            }
+        }
+        if (!getClassName || !getStackTrace) {
+            return;
+        }
+        for (AbstractInsnNode insn = method.instructions.getFirst(); insn != null; insn = insn.getNext()) {
+            if (insn.getOpcode() == Opcodes.NEW) {
+                TypeInsnNode typeInsn = (TypeInsnNode) insn;
+                if (typeInsn.desc.endsWith("Exception") || typeInsn.desc.endsWith("Error")) {
+                    typeInsn.desc = "java/lang/RuntimeException";
+                }
+            } else if (insn instanceof MethodInsnNode) {
+                MethodInsnNode methodInsn = (MethodInsnNode) insn;
+                if (methodInsn.owner.endsWith("Exception") || methodInsn.owner.endsWith("Error")) {
+                    methodInsn.owner = "java/lang/RuntimeException";
                 }
             }
         }
     }
 
-    private int cleanup(Set<MethodNode> methods) {
+    private int cleanup(Set<MethodNode> toRemove) {
+        //remove methods from removal set which are still called
+        classNodes().forEach(node -> node.methods.forEach(methodNode -> {
+            for (AbstractInsnNode insn : methodNode.instructions) {
+                if (insn.getOpcode() != INVOKESTATIC) {
+                    continue;
+                }
+                MethodInsnNode m = (MethodInsnNode) insn;
+                ClassNode owner = classes.get(m.owner);
+                if (owner == null) {
+                    continue;
+                }
+                MethodNode mNode = owner.methods.stream().filter(mn -> mn.name.equals(m.name) && mn.desc.equals(m.desc)).findFirst().orElse(null);
+                toRemove.remove(mNode);
+            }
+        }));
         AtomicInteger count = new AtomicInteger(0);
         classNodes().forEach(classNode -> {
-            if (classNode.methods.removeIf(methods::contains)) {
-                count.getAndIncrement();
+            for (Iterator<MethodNode> it = classNode.methods.iterator(); it.hasNext(); ) {
+                MethodNode methodNode = it.next();
```

Comment:
Why are you iterating like this

Reply:
Because I use it.remove()

Reply:
use removeIf?

Reply:
That was how it worked before, but it did not count correctly. It counted amount of edited classes and not amount of removed methods.

Reply:
> That was how it worked before, but it did not count correctly. It counted amount of edited classes and not amount of removed methods.

Huh.  I'm honestly not sure why that happens.  Will make a note to look into that at some point.

Reply:
> 
> 
> > That was how it worked before, but it did not count correctly. It counted amount of edited classes and not amount of removed methods.
> 
> Huh. I'm honestly not sure why that happens. Will make a note to look into that at some point.

Uh, isn't that obvious?

The previous code uses classNode.methods.removeIf which returns true if at least one method was removed, and then it increased count by one. The new style increases count every time we remove a method.

Reply:
Ah wait, I see.  Yes, it is trivial.  Looks like I was just being 2iq at the time.